pegged/peg.d

```diff
@@ -49,7 +49,7 @@ package string stringified(string inp) @safe
             "\t" : `"\t"`,
 
             `"` : `"\""`,
-            "`" : q"("`")",
+            "`" : q{("`")},
```

Comment:
You changed this from a [delimited string](https://dlang.org/spec/lex.html#delimited_strings) to a [token string](https://dlang.org/spec/lex.html#token_strings), which broke the test.

Reply:
A long while ago I posted ANTLR runtime to D. But I think peg fits better to D.
But what I like with ANTLR is that you can do channels and I was thinking of some thinking similar in pegged.
But for now, I think we should close my suggestion for this change now. 
I will try to make a better proposal so we can have some context ParseTree.

Reply:
OK, you can close the PR yourself if you wish. I don't know what channels are in parsers, but I look forward to see your work.

Reply:
Hej Veelo.
I have made an updated, so it should be possible to make a costume ParseTree.
It parses the unittest, but it is not tested yet. I will use it in my other project now so I will and in the process make a test for it.

Note.
Channel is used when you have syntax groups.
Like is C.
You have comments, macros, code, and maybe more.
So with channels, you can write a grammar that adds a channel to a part of the grammar.
When you traverse the tree you can select which part to traverse. Ex if you don' what to look at the comment you can filter to about.

Reply:
So what Pegged does is that it builds a parse tree as it processes the input. After that the user traverses the parse tree in order to interpret it and extract data from it. So you could say that there are two stages in a Pegged parsing process. At what stage would you select a particular channel? I mean, does selecting a channel affect the parse tree? If it is in the first stage, it seems to me to be somewhat like selectively dropping nodes. That should be possible today using semantic actions. If it is in the second stage, it would be up to the user to implement this, no?
There is also the feature of composable grammars, so you can have a grammar for comments and a grammar for code.

Reply:
Hej Veelo.
I have added the option to customize the ParseTree.
At first glance, it seems that the code has changed a lot but in principle, I have just added some mixin templates and some introspection functions.
I have tried not to destroy your code ;-).
I have also changed so most of the code is now @safe.

In the makefile I have added a
make test-all -k
Which runs a dub test on all the examples.
The same tests parse for this branch as for the master.

 
And I have added a makefile inside the pegged/dev to regenerate the parser.d

I will now continue to use this version in more project and I will post you the result and some ex